### PR TITLE
Feature/font generator

### PR DESCRIPTION
**Pull request by cvasqu09**
Added the font generator, atlas for cutting out textures for buttons, and modified the buttons to properly overlay text on them. Added the banner to all of the screens.

### CODE REVIEW

File: core/src/main/java/Scenes/BaseScreen.java [hunk 1]
```diff
@@ -18,18 +26,42 @@
...
+  protected BitmapFont bannerText;
+  protected Skin buttonSkin;
+  private FreeTypeFontGenerator generator;
+  private FreeTypeFontParameter parameter;
 
   // This is the native screen size that will be the reference for everything placed on the screen.
   protected static final float WORLD_WIDTH = 480;
   protected static final float WORLD_HEIGHT = 800;
 
+  // Used to calculate where to draw the text in the banner so that it is centered
```
Comment (jmtaber129) [inline core/src/main/java/Scenes/BaseScreen.java]: Missing period

File: core/src/main/java/Scenes/Leaderboards.java [hunk 2]
```diff
@@ -43,16 +47,26 @@ public void render(float delta) {
...
     spriteBatch.draw(new Sprite(leaderboardsMockUp), 65, 300);
     spriteBatch.end();
   }
 
   /**
+   * Creates all of the  buttons that will be drawn to the screen.
+   */
+  @Override
+  protected void createButtons() {
+    // Creates the back button on the banner to go back a screen.
```
Comment (jmtaber129) [inline core/src/main/java/Scenes/Leaderboards.java]: Move this description to where 'backButton' is declared.

File: core/src/main/java/Scenes/Lobby.java [hunk 3]
```diff
@@ -48,35 +44,50 @@ public void render(float delta) {
...
+  protected void createButtons() {
+    // Creates the back button to go back to the previous screen at the given position.
+    backButton = new ButtonActor(new Texture(Gdx.files.internal("black-back-arrow.png")), 400, 735);
+    TextButtonStyle style = new TextButtonStyle();
+    // Sets the skin for when the button is not pressed and when it is. The argument that is passed
+    // is searched for in the atlas within the buttonSkin object.
+    style.up = buttonSkin.getDrawable("heavy-sat-yellow-246x46");
+    style.down = buttonSkin.getDrawable("heavy-sat-yellow-246x46");
+    style.font = generateNewFont("Rampung.ttf", 30, Color.BLACK);
+    // Creates the play button at the specificed location with the given style.
```
Comment (jmtaber129) [inline core/src/main/java/Scenes/Lobby.java]: Move this button description to where it's declared.

File: core/src/main/java/Scenes/MainMenu.java [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
 import com.mygdx.game.FilePlayMain;
 
+
 import UIElements.ButtonActor;
 
 /**
- * TODO(Chris): Refactor world width, height, batch, and ratios into Game class. TODO(Chris):
- * Include relative offsets and spacing. MainMenu will display the main menu screen of the game with
- * buttons that transition them into appropriate screens from the main menu.
+ * MainMenu will display the main menu screen of the game with buttons that transition them into
```
Comment (jmtaber129) [inline core/src/main/java/Scenes/MainMenu.java]: I think this should be in present tense: "MainMenu displays..."

File: core/src/main/java/Scenes/Leaderboards.java (hunk 2 above)
Comment (cvasqu09) [inline core/src/main/java/Scenes/Leaderboards.java]: Fixed

File: core/src/main/java/Scenes/Lobby.java (hunk 3 above)
Comment (cvasqu09) [inline core/src/main/java/Scenes/Lobby.java]: Fixed

File: core/src/main/java/Scenes/MainMenu.java (hunk 4 above)
Comment (cvasqu09) [inline core/src/main/java/Scenes/MainMenu.java]: Fixed

**Comment by jmtaber129** [comment]
Not sure how I feel about the back button being on the right-hand side, and I don't think the back button should be present on the main menu (what would we be going back to?), but these are things we can discuss on Tuesday.

LGTM

**Pull request merged**